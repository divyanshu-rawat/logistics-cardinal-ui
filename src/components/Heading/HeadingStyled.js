import styled from 'styled-components';

const TYPES = {
  h1: {
    fontSize: 'giga',
    lineHeight: 'kilo',
  },
  h2: {
    fontSize: 'mega',
    lineHeight: 'byte',
  },
  h3: {
    fontSize: 'kilo',
    lineHeight: 'byte',
  },
  h4: {
    fontSize: 'byte',
    lineHeight: 'bit',
  },
  h5: {
    fontSize: 'bit',
    lineHeight: 'bit',
  },
};

const HeadingStyled = styled('h3')`
  font-size: ${({ theme, as }) => theme.fontSize[(TYPES[as] || {}).fontSize]};
  font-weight: ${({ theme, bold }) =>
    theme.fontWeight[bold ? 'semiBold' : 'regular']};
  line-height: ${({ theme, as }) =>
    theme.lineHeight[(TYPES[as] || {}).lineHeight]};
  margin: 0 0 ${({ theme, noMargin }) => (noMargin ? 0 : theme.spacings.kilo)};
`;

export default HeadingStyled;
