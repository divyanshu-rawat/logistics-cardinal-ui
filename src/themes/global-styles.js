import { createGlobalStyle } from 'styled-components';
import 'normalize.css';

export const GlobalStyles = createGlobalStyle`
  ${({ theme, injectGlobal }) => `
    *,
    *::before,
    *::after {
      box-sizing: border-box;
    }

    html {
      box-sizing: border-box;
      line-height: 1.15;
      -webkit-text-size-adjust: 100%;
      -ms-text-size-adjust: 100%;
      -ms-overflow-style: scrollbar;
      -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
      [type='button'] {
        appearance: none;
      }
    }

    body {
      margin: 0;
      background-color: ${theme.bodyBackground};
      color: ${theme.textColor};
      font-family: ${theme.textFontFamily};
      font-size: ${theme.textFontSize};
      font-weight: ${theme.textFontWeight};
      line-height: ${theme.lineHeightBase};
    }

    fieldset {
      min-width: 0;
      padding: 0;
      margin: 0;
      border: 0;
    }

    .Drawer {
      &--is-open {
        overflow: hidden;
      }
    }

    .ReactModal {
      &__Body--open {
        overflow: hidden;
      }
      &__Overlay {
        position: fixed;
        top: 0;
        left: 0;
        right: 0;
        bottom: 0;
        transition: opacity 200ms ease-in-out;
        height: auto;
        overflow: auto;
        z-index: ${theme.zIndex.modalOverlay};
        background-color: ${theme.overlayColor};
        &--after-open {
          opacity: 1;
        }
        &--before-close {
          opacity: 0;
        }
      }
      &__Content {
        margin: 10em auto;
        position: relative;
        left: auto;
        right: auto;
        top: auto;
        bottom: auto;
        transform: none;
        opacity: 0;
        transition: opacity 200ms ease-in-out;
        min-height: ${theme.modalMinHeight};
        min-width: ${theme.modalMinWidth};
        width: ${theme.modalWidth};
        max-width: ${theme.modalMaxWidth};
        background-color: ${theme.modalBackgroundColor};
        border: 1px ${theme.modalBorderStyle} ${theme.modalBorderColor};
        box-shadow: ${theme.modalBoxShadow};
        border-radius: ${theme.modalBorderRadius};
        outline: ${theme.modalOutline};
        z-index: ${theme.zIndex.modalContent};
        &--after-open {
          opacity: 1;
        }
        &--before-close {
          opacity: 0;
        }
      }
    }


    ${injectGlobal && injectGlobal(theme)};
  `}
`;
