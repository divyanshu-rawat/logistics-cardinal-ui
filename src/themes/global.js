import { injectGlobal } from 'styled-components';

import 'react-select/dist/react-select.css';
import 'react-select-plus/dist/react-select-plus.css';

/* eslint no-unused-expressions: 0 */
injectGlobal`
  @import url('https://fonts.googleapis.com/css?family=Open+Sans:300,400,600,700,800');

  // used to overwrite bootstrap's background-color
  table[class^="css-"] {
    background: white;
    td[class^="css-"] {
      white-space: initial;
    }
  }
`;
