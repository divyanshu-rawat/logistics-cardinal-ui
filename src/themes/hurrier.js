const white = '#fff';
const black = '#000';

const primary = {
  primary100: '#db0062',
  primary200: '#375f8a',
};

const greys = {
  grey100: '#96a2b0',
  grey200: '#3f454d',
  grey300: '#343a42',
  grey400: '#292d33',
};

const greens = {
  green100: '#c6e5cb',
  green300: '#97e2a3',
  green500: '#62de76',
  green700: '#49b85b',
  green900: '#2f903e',
};

const reds = {
  red100: '#ffe6e1',
  red300: '#ffaf9f',
  red500: '#ff7559',
  red700: '#d55a41',
  red900: '#a73d28',
};

const yellows = {
  yellow100: '#f2e5cb',
  yellow300: '#f7da9f',
  yellow500: '#ffc859',
  yellow700: '#d4a546',
  yellow900: '#ab8433',
};

const misc = {
  bodyBackground: '#fff',
  mainBackground: '#ccc',
  textColor: white,
  danger: reds.red500,
  success: greens.green700,
  warning: yellows.yellow700,
};

const colors = {
  white,
  black,
  ...primary,
  ...greys,
  ...greens,
  ...reds,
  ...yellows,
  ...misc,
};

const spacings = {
  bit: '4px',
  byte: '8px',
  kilo: '12px',
  mega: '16px',
  giga: '24px',
  tera: '32px',
  peta: '40px',
  exa: '48px',
  zetta: '56px',
};

const borderRadius = {
  kilo: '1px',
  mega: '4px',
  giga: '5px',
};

const borderWidth = {
  kilo: '1px',
  mega: '2px',
};

const fontFamily = {
  default: '"Open Sans", sans-serif',
};

const fontWeight = {
  regular: '400',
  bold: '700',
};

export default {
  colors,
  spacings,
  borderRadius,
  borderWidth,
  fontFamily,
  fontWeight,
};
