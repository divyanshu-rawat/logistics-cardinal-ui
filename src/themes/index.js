export rooster from './rooster';
export standard from './standard';
export hurrier from './hurrier';
