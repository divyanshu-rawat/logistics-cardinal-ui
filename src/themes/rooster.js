const white = '#fff';
const black = '#000';

const primary = {
  primary100: '#3f9bfd',
  primary200: '#375f8a',
  primary300: '#324e70',
  primary400: '#96a2b0',
  primary500: '#484e58',
};

const greys = {
  grey100: '#96a2b0',
  grey200: '#3f454d',
  grey300: '#343a42',
  grey400: '#292d33',
  grey500: '#40464f',
};

const greens = {
  green100: '#c6e5cb',
  green300: '#97e2a3',
  green500: '#62de76',
  green700: '#49b85b',
  green900: '#2f903e',
};

const reds = {
  red100: '#ffe6e1',
  red300: '#ffaf9f',
  red500: '#ff7559',
  red700: '#d55a41',
  red900: '#a94442',
};

const yellows = {
  yellow100: '#f2e5cb',
  yellow300: '#f7da9f',
  yellow500: '#ffc859',
  yellow700: '#d4a546',
  yellow900: '#ab8433',
};

const misc = {
  bodyBackground: greys.grey400,
  mainBackground: '343941',
  textColor: white,
  danger: reds.red900,
  success: greens.green700,
  warning: yellows.yellow700,
};

const colors = {
  white,
  black,
  ...primary,
  ...greys,
  ...greens,
  ...reds,
  ...yellows,
  ...misc,
};

const spacings = {
  bit: '1px',
  byte: '5px',
  kilo: '10px',
  mega: '15px',
  giga: '20px',
};

const composedSpacings = {
  byte: `${spacings.kilo} ${spacings.mega}`,
};

const grid = {
  outerGutter: '10px',
};

const borderRadius = {
  kilo: '1px',
  mega: '4px',
  giga: '5px',
};

const borderWidth = {
  bit: '1px',
  byte: '2px',
};

const fontFamily = {
  default: '"Open Sans", sans-serif',
  color: colors.white,
};

const fontSize = {
  bit: '11px',
  byte: '13px',
  kilo: '15px',
  mega: '18px',
  giga: '24px',
};

const lineHeight = {
  bit: 1.153,
  byte: 1.333,
  kilo: 1.5,
};

const fontWeight = {
  regular: 400,
  semiBold: 600,
  bold: 700,
};

export default {
  colors,
  grid,
  spacings,
  composedSpacings,
  borderRadius,
  borderWidth,
  fontFamily,
  fontWeight,
  fontSize,
  lineHeight,
};
