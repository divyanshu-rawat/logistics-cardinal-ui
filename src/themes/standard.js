// Required Fields
export const aliases = {
  /**
   * Text Variables
   */
  textColor: 'black',
  textFontSize: '12px',
  textFontWeight: 300,
  textFontFamily: '',
  textSpacing: '10px',
  textFontSizeXSmall: '',
  textFontSizeSmall: '',
  textFontSizeLarge: '',
  textFontSizeXLarge: '',
  /**
   * Heading Variables
   */
  h1FontSize: '32px',
  h1LineHeight: '',
  h2FontSize: '',
  h2LineHeight: '',
  h3FontSize: '',
  h3LineHeight: '',
  h4FontSize: '',
  h4LineHeight: '',
  h5FontSize: '',
  h5LineHeight: '',
  headingSpacing: '',
  headingFontWeightLight: '',
  headingFontWeightBold: '',
  /**
   * Color Variables
   */
  bodyBackground: 'white',
  mainBackground: '',
  primaryColor: '#d70f64',
  dangerColor: 'red',
  successColor: 'green',
  warningColor: 'yellow',
  infoColor: 'cyan',
  overlayColor: 'rgba(25, 29, 35, 0.82)',
  /**
   * Line Height Variables
   */
  lineHeightBase: 1.5,
  /**
   * Input and Select Variables
   */
  inputDefaultHeight: '40px',
  inputPlaceholderColor: '',
  inputBackgroundColor: '',
  inputDisabledBackgroundColor: '',
  inputBorderColor: '',
  inputErrorBorderColor: '',
  inputBorderColorHover: '',
  inputBorderRadius: '',
  inputTextColor: '',
  inputLineHeight: '',
  inputFontSize: '',
  inputFontWeight: '',
  inputBoxShadow: '',
  inputPadding: '',
  // Select
  selectItemFocus: '',
  selectItemDisabledColor: '',
  selectItemDisabledHoverBackgroundColor: '',
  selectMultiTagBackgroundColor: '',
  selectMultiTagTextColor: '',
  selectSingleTextColor: '',
  /**
   * Input Feedback Variables
   */
  inputFeedbackSpacing: '',
  inputFeedbackFontWeight: '',
  inputFeedbackFontSize: '',
  /**
   * Button Variables
   */
  buttonFontWeight: '',
  buttonFontColor: '',
  buttonBorderWidth: '',
  buttonFocusBoxShadow: '',
  buttonDisabledOpacity: 0.65,
  buttonActiveBoxShadow: 'inset 0 3px 5px rgba($black, .125)',
  buttonPaddingY: '10px',
  buttonPaddingYXSmall: '1px',
  buttonPaddingYLarge: '18px',
  buttonPaddingYSmall: '6px',
  buttonPaddingSmall: '9px',
  buttonPaddingX: '15px',
  buttonPaddingXSmall: '5px',
  buttonPaddingXLarge: '27px',
  buttonFontSize: '12px',
  buttonFontSizeSmall: '11px',
  buttonFontSizeLarge: '15px',
  buttonLineHeight: '',
  buttonLineHeightSmall: 1.5,
  buttonLineHeightLarge: 1.333,
  buttonBorderRadius: '',
  /**
   * Radio Button Variables
   */
  radioSpacingRight: '',
  radioFontWeight: '',
  radioInactiveBorderColor: '#ccc',
  /**
   * Indicator Radio Button Variables
   */
  indicatorColor: '',
  indicatorSpacingRight: '',
  indicatorCheckedColor: '',
  /**
   * Label Variables
   */
  labelTextColor: '',
  labelTextColorError: '',
  labelFontWeight: '',
  labelFontSize: '',
  labelBottomSpacing: '',
  /**
   * Drawer Variables
   */
  drawerBackgroundColor: '',
  drawerBoxShadow: '',
  drawerPadding: '',
  /**
   * Spacing Variables
   */
  spacing: '',
  /**
   * Spinner Variables
   */
  spinnerOverlayBackgroundColor: 'rgba(52, 58, 66, 0.8)',
  /**
   * Switcher Variables
   */
  switcherSpacingRight: '',
  switcherBorderColor: '',
  switcherBoxShadow: '',
  /**
   * Tabs - Horizontal - Variables
   */
  horizontalTabButtonBorderRadius: '',
  horizontalTabButtonBackgroundColor: '',
  horizontalTabButtonColor: '',
  horizontalTabButtonColorActive: '',
  horizontalTabButtonSpacingRight: '',
  horizontalTabButtonPadding: '',
  horizontalTabContainerPadding: '',
  horizontalTabContainerBackgroundColor: '',
  /**
   * Tabs - Vertical - Variables
   */
  verticalTabButtonBackgroundColor: '',
  verticalTabButtonSpacingRight: '',
  verticalTabButtonPadding: '',
  verticalTabButtonActiveBackgroundColor: '',
  verticalTabSidebarBackgroundColor: '',
  verticalTabContainerPadding: '',
  verticalTabContainerBackgroundColor: '',
  /**
   * TimeSpanDisplay - Variables
   */
  timeSpanDisplayTextColor: '',
  /**
   * Tooltip - Variables
   */
  tooltipBorderRadius: '',
  tooltipBackgroundColor: '',
  tooltipBorderColor: '',
  tooltipTextColor: '',
  tooltipFontSize: '',
  tooltipHoverOpacity: 0.9,
  /**
   * Transitions
   */
  transitions: {
    overlayTransition: 'opacity 0.3s ease-out, visibility 0.3s ease-out',
  },
  /**
   * Modal
   */
  modalBackgroundColor: '',
  modalBorderColor: '',
  modalBorderStyle: '',
  modalBoxShadow: '',
  modalBorderRadius: '',
  modalOutline: 0,
  modalWidth: '80%',
  modalMaxWidth: '60rem',
  modalMinWidth: '20rem',
  minHeight: '10rem',
  modalWidthSmall: '400px',
  /**
   * Z-Index
   */
  zIndex: {
    select: 1090,
    tooltip: 2020,
    spinnerOverlay: 2040,
    drawerOverlay: 2040,
    drawerContent: 2041,
    spinner: 2041,
    navbar: 2030,
    loadingBar: 3000,
    modalOverlay: 2050,
    modalContent: 2060,
  },
  /**
   * Media Query Breakpoints
   */
  breakpoints: {
    default: {
      minWidth: 768,
    },
    largeDesktop: {
      minWidth: 1200,
    },
    desktop: {
      minWidth: 992,
      maxWidth: 1200,
    },
    mobile: {
      minWidth: 0,
      maxWidth: 768,
    },
    tablet: {
      minWidth: 768,
      maxWidth: 992,
    },
  },
  grid: {
    outerGutter: '10px',
  },
};

export const StandardTheme = {
  ...aliases,
};
