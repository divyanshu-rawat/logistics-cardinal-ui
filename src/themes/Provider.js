import React, { Fragment } from 'react';
import { ThemeProvider } from 'styled-components';
import PropTypes from 'prop-types';

import { standard } from '.';
import { GlobalStyles } from './global-styles';

const CardinalThemeProvider = ({ theme, injectGlobal, children }) => {
  const mergedTheme = {
    ...standard,
    ...theme,
  };

  return (
    <ThemeProvider theme={mergedTheme}>
      <Fragment>
        <GlobalStyles injectGlobal={injectGlobal} />
        {children}
      </Fragment>
    </ThemeProvider>
  );
};

CardinalThemeProvider.propTypes = {
  children: PropTypes.node.isRequired,
  injectGlobal: PropTypes.func,
  theme: PropTypes.object,
};

export { CardinalThemeProvider };
