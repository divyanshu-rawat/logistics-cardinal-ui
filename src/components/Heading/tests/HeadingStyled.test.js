import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import HeadingStyled from '../HeadingStyled';
import { RoosterTheme } from '../../../themes/rooster';

const renderedComponent = ({ as = 'h3', bold = false, noMargin = false }) =>
  shallow(
    <HeadingStyled theme={RoosterTheme} as={as} bold={bold} noMargin={noMargin}>
      Almighty Corinthians!
    </HeadingStyled>,
  );

describe('<HeadingStyled />', () => {
  it('should render the component with the default props', () => {
    const tree = toJson(renderedComponent({}));
    expect(tree).toMatchSnapshot();
  });

  it('should render the component with different props', () => {
    const tree = toJson(
      renderedComponent({
        as: 'h4',
        bold: true,
        noMargin: true,
      }),
    );
    expect(tree).toMatchSnapshot();
  });

  it('should still rendering the component even if the element doesnt match the list', () => {
    const tree = toJson(
      renderedComponent({
        as: 'span',
        bold: true,
        noMargin: true,
      }),
    );
    expect(tree).toMatchSnapshot();
  });
});
