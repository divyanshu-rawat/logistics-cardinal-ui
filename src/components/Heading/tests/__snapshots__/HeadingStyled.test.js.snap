// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeadingStyled /> should render the component with different props 1`] = `
<h3
  as="h4"
  className="sc-bdVaJa dcLdYj"
>
  Almighty Corinthians!
</h3>
`;

exports[`<HeadingStyled /> should render the component with the default props 1`] = `
<h3
  as="h3"
  className="sc-bdVaJa kvBkri"
>
  Almighty Corinthians!
</h3>
`;

exports[`<HeadingStyled /> should still rendering the component even if the element doesnt match the list 1`] = `
<h3
  as="span"
  className="sc-bdVaJa ftJEaA"
>
  Almighty Corinthians!
</h3>
`;
